api/src/main/java/io/grpc/AdminInterface.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Admin Interface provides a class of services for exposing the overall state of all gRPC
+ * activity in a given binary. It aims to be a convenient API that automatically determines
+ * which specific admin services are relevant.
+ */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/7929")
+@ThreadSafe
+public final class AdminInterface {
+  private static final int DEFAULT_CHANNELZ_MAX_PAGE_SIZE = 100;
+  private static final String CHANNELZ_CLASS = "io.grpc.services.ChannelzService";
+  private static final String CSDS_CLASS = "io.grpc.xds.CsdsService";
+  private static final Logger logger = Logger.getLogger(AdminInterface.class.getName());
+  private static List<ServerServiceDefinition> standardServices;
```

Comment:
I don't see much point in caching the result. It seems it'd be rare for `getStandardServices()` to be called _twice_, much less enough times such that a cache is useful.

Reply:
Fixed.